samples/extensions/logic_op_dynamic_state/logic_op_dynamic_state.cpp

```diff
@@ -0,0 +1,726 @@
+#include <string>
+#include <vector>
+
+#include "logic_op_dynamic_state.h"
+
+#include "gltf_loader.h"
+
+/* Initialize the static variable containing a vector of logic operations' labels for GUI */
+std::vector<std::string> LogicOpDynamicState::GUI_settings::logic_op_names =
+    LogicOpDynamicState::GUI_settings::init_logic_op_names();
+
+LogicOpDynamicState::LogicOpDynamicState()
+{
+	title = "Logic Operations Dynamic State";
+
+	/* Extensions required for dynamic logic operations */
+	add_instance_extension(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME);
+	add_device_extension(VK_EXT_EXTENDED_DYNAMIC_STATE_2_EXTENSION_NAME);
+	add_device_extension(VK_EXT_EXTENDED_DYNAMIC_STATE_EXTENSION_NAME);
+}
+
+LogicOpDynamicState::~LogicOpDynamicState()
+{
+	if (device)
+	{
+		uniform_buffers.common.reset();
+		uniform_buffers.baseline.reset();
+
+		vkDestroySampler(get_device().get_handle(), textures.envmap.sampler, VK_NULL_HANDLE);
+		textures = {};
+
+		vkDestroyPipeline(get_device().get_handle(), pipeline.baseline, VK_NULL_HANDLE);
+		vkDestroyPipeline(get_device().get_handle(), pipeline.background, VK_NULL_HANDLE);
+
+		vkDestroyPipelineLayout(get_device().get_handle(), pipeline_layouts.baseline, VK_NULL_HANDLE);
+		vkDestroyPipelineLayout(get_device().get_handle(), pipeline_layouts.background, VK_NULL_HANDLE);
+
+		vkDestroyDescriptorSetLayout(get_device().get_handle(), descriptor_set_layouts.baseline, VK_NULL_HANDLE);
+		vkDestroyDescriptorSetLayout(get_device().get_handle(), descriptor_set_layouts.background, VK_NULL_HANDLE);
+	}
+}
+
+/**
+ * 	@fn bool LogicOpDynamicState::prepare(vkb::Platform &platform)
+ * 	@brief Configuring all sample specific settings, creating descriptor sets/pool, pipelines, generating or loading models etc.
+ */
+bool LogicOpDynamicState::prepare(vkb::Platform &platform)
+{
+	if (!ApiVulkanSample::prepare(platform))
+	{
+		return false;
+	}
+
+	/* Set up camera properties */
+	camera.type = vkb::CameraType::LookAt;
+	camera.set_position({2.0f, -4.0f, -10.0f});
+	camera.set_rotation({-15.0f, 190.0f, 0.0f});
+	camera.set_perspective(60.0f, static_cast<float>(width) / static_cast<float>(height), 256.0f, 0.1f);
+
+	load_assets();
+	model_data_creation();
+	prepare_uniform_buffers();
+	create_descriptor_pool();
+	setup_descriptor_set_layout();
+	create_descriptor_sets();
+	create_pipeline();
+	build_command_buffers();
+
+	prepared = true;
+	return true;
+}
+
+/**
+ * 	@fn void LogicOpDynamicState::create_render_context(vkb::Platform &platform)
+ * 	@brief Setting custom surface format priority list to required VK_FORMAT_B8G8R8A8_UNORM format
+ */
+void LogicOpDynamicState::create_render_context(vkb::Platform &platform)
+{
+	/* UNORM surface is required for logic operations */
+	auto surface_priority_list = std::vector<VkSurfaceFormatKHR>{
+	    {VK_FORMAT_B8G8R8A8_UNORM, VK_COLORSPACE_SRGB_NONLINEAR_KHR},
+	};
+	render_context = platform.create_render_context(*device.get(), surface, surface_priority_list);
```

Comment:
get here is redundant on the smart pointer.

Reply:
changed to "get_device()"